### ISSUE TITLE
(Bug): Returning NAN instead of 0 for dislikes

### ISSUE THREAD

**Issue by KJ002** [Anarios/return-youtube-dislike#142]
### Browser

Firefox

### Browser Version

Mozilla Firefox 94.0

### Extension or Userscript?

Extension

### Extension/Userscript Version

1.0.0.2

### Video link where you see the problem

https://www.youtube.com/watch?v=CP4jHWSdmpU

### What happened?

The dislikes displayed `NAN` where it should display `0`.
![nan](https://user-images.githubusercontent.com/36408549/144327432-b645ed39-ce14-4197-bb08-241a42c24ab2.png)

**Comment by Paulmski** [comment]
Hi there, I was unable to reproduce with the following:
Fedora 35
Firefox
Mozilla Firefox 94.0
Extension 1.0.0.2

Video:
[https://www.youtube.com/watch?v=lz8-fdVtox8]()

Result:

![Screenshot from 2021-12-01 19-23-27](https://user-images.githubusercontent.com/78445698/144335761-44de14e4-f43f-4aa3-90db-36db1317faff.png)


Do you have any other information that might help reproduce? Does this happen with other videos?

**Comment by TheJuki** [comment]
https://www.youtube.com/watch?v=xAlQ-UF6xgw

I saw it on this video. Refreshing showed 3 dislikes. I think because I was the only one to watch with the extension first.

**Comment by Paulmski** [comment]
Thanks! I can reproduce this, seems to be as you say. Only videos that have never been queried for have this behavior, or so it seems.

**Comment by samir-dahal** [comment]
unable to reproduce
![image](https://user-images.githubusercontent.com/39849021/144442471-df95136a-1052-4305-a106-33a438437eb9.png)

**Comment by Quendi6** [comment]
I am on Firefox, while the page is loading I see the dislikes count, until the page has finished loading. Then I see NAN, for each video.

**Comment by Paulmski** [comment]
> [return youtube dislike]: response from api:
return-youtube-dislike.script.js:7 [return youtube dislike]: {"type":"https://tools.ietf.org/html/rfc7231#section-6.5.4","title":"Not Found","status":404,"traceId":"00-823929c12977591ace4e4d7db96a7c04-a9f4e8aa50e8ea0d-00"}
watch:1 Error handling response: TypeError: Cannot read properties of undefined (reading 'toLocaleString')
at createRateBar (chrome-extension://gebbhagfogifgggkldgodflihgfeippi/return-youtube-dislike.script.js:198:44)
at chrome-extension://gebbhagfogifgggkldgodflihgfeippi/return-youtube-dislike.script.js:105:11 


Judging from the console log it seems like this would be better solved on the back end. It appears like the backend isn't properly handling a query that has never occurred before. It would be nice to have access to the backend to create a more elegant solution than just patching the extension to mask the error. Or even just to investigate this bug further.

**Comment by Quendi6** [comment]
In first place it load old values:

```
[return youtube dislike]: response from api: return-youtube-dislike.script.js:6:13
[return youtube dislike]: {"id":"fe-vBlVUcPM","dateCreated":"2021-11-24T04:54:37.99698Z","likes":98430,"dislikes":846,"rating":4.965913211652363,"viewCount":1314417}
```

Then in a second attempt, it refresh likes / dislikes / viewCount values:

```
[return youtube dislike]: response from youtube: return-youtube-dislike.script.js:6:13
[return youtube dislike]: {"likes":98890,"dislikes":null,"viewCount":1337167}
```

**Comment by Paulmski** [comment]
@Quendi6 Yeah, dealing with null values client-side isn't ideal. Seems like this will have to wait until the backend is public.

**Comment by Robin5605** [comment]
We could temporarily set it to 0 if the dislikes is returned at null, but yeah having access to the backend code would be very helpful.

**Comment by KendallDoesCoding** [comment]
> ### Browser
> Firefox
> 
> ### Browser Version
> Mozilla Firefox 94.0
> 
> ### Extension or Userscript?
> Extension
> 
> ### Extension/Userscript Version
> 1.0.0.2
> 
> ### Video link where you see the problem
> https://www.youtube.com/watch?v=CP4jHWSdmpU
> 
> ### What happened?
> The dislikes displayed `NAN` where it should display `0`. ![nan](https://user-images.githubusercontent.com/36408549/144327432-b645ed39-ce14-4197-bb08-241a42c24ab2.png)

I am facing the same issue #152

**Comment by Anarios** [comment]
Fixed in 1.1.0.0 Released to FF, in approval for chrome (coming tommorow)

**Comment by KendallDoesCoding** [comment]
> Fixed in 1.1.0.0 Released to FF, in approval for chrome (coming tommorow)

Thanks!
